modAionImpl/src/org/aion/zero/impl/sync/handler/BlockPropagationHandler.java

```diff
@@ -48,12 +50,18 @@
 
     private static final byte[] genesis = CfgAion.inst().getGenesis().getHash();
 
+    private final byte apiVersion;
+
+    private final List<AionTransaction> pendingTransactions;
+
     public BlockPropagationHandler(
             final int cacheSize,
             final IAionBlockchain blockchain,
             final IP2pMgr p2pManager,
             BlockHeaderValidator<A0BlockHeader> headerValidator,
-            final boolean isSyncOnlyNode) {
+            final boolean isSyncOnlyNode,
+            final byte apiVersion,
+            final List<AionTransaction> pendingTransactions) {
```

Comment:
what's the reason to add this?

Reply:
processNewBlock function inside BlockPropagationHandler calls the ResStatus constructor

Reply:
the handler should pass the txpool interface instead of the list of tx.